charts/consul/templates/gateway-resources-configmap.yaml

```diff
@@ -1,4 +1,9 @@
 {{- if .Values.connectInject.enabled }}
+
+# Validation
+# For meshGateway.wanAddress, static must be set if source is "Static"
+{{if (and (eq .Values.meshGateway.wanAddress.source "Static") (eq .Values.meshGateway.wanAddress.static ""))}}{{fail ".meshGateway.wanAddress.static must be set to a value if .meshGateway.wanAddress.source is Static"}}{{ end }}
```

Comment:
I added validation here that will block the invalid config of a "Static" address with no address set.